options.go

```diff
@@ -0,0 +1,104 @@
+package notify
+
+import "encoding/json"
+
+type PayloadItem struct {
+	field   string
+	message interface{}
+}
+
+type Payload []PayloadItem
+
+func (payload Payload) MarshalJSON() ([]byte, error) {
+	dict := map[string]interface{}{}
+
+	for _, item := range payload {
+		dict[item.field] = item.message
+	}
+
+	return json.Marshal(dict)
+}
+
+// Personalisation is a slice of structs used to define placeholder values in a template,
+// such as name or reference number.
+// The struct should be structured such that they key is the name of the value in your template, and the value is what you expect to be substituted in the message.
+type Personalisation []struct{ Key, Value string }
+
+func (personalisation Personalisation) UpdatePayload(payload Payload) Payload {
+	dict := map[string]string{}
+
+	for _, item := range personalisation {
+		dict[item.Key] = item.Value
+	}
+
+	return append(payload, PayloadItem{"personalisation", dict})
+}
+
+func (personalisation Personalisation) UpdateSMSPayload(p Payload) Payload {
+	return personalisation.UpdatePayload(p)
+}
+
+func (personalisation Personalisation) UpdateEmailPayload(p Payload) Payload {
+	return personalisation.UpdatePayload(p)
+}
+
+// Reference is a unique identifier you create. It identifies a single unique notification or a batch of notifications.
+func Reference(referenceID string) CommonOption {
+	return CommonOption{
+		UpdatePayloadFunc(func(payload Payload) Payload {
+			return append(payload, PayloadItem{"reference", referenceID})
+		}),
+	}
+}
+
+// EmailReplyToID is the id of the reply-to address to receive replies from users.
+func EmailReplyToID(address string) SendEmailOption {
+	return UpdatePayloadFunc(func(payload Payload) Payload {
+		return append(payload, PayloadItem{"email_reply_to_id", address})
+	})
+}
+
+// SMSSenderID is a unique identifier for the sender of a text message.
+func SMSSenderID(senderID string) SendSMSOption {
+	return UpdatePayloadFunc(func(payload Payload) Payload {
+		return append(payload, PayloadItem{"sms_sender_id", senderID})
+	})
+}
+
+type PayloadUpdater interface {
```

Comment:
@dalehurwitz can you please unexport `PayloadUpdater` to `payloadUpdater` and `UpdatePayload` to `updatePayload`. Double check that it works with them unexported but pretty sure it does.

Reply:
@dalehurwitz also `UpdatePayloadFunc` to `updatePayloadFunc`.

Reply:
@dalehurwitz also `Payload` to `payload` and `PayloadItem` to `payloadItem` and also unexport all of:

```
UpdateEmailPayload
UpdatePayload
UpdateSMSPayload
```